optimum/neuron/utils/hub_neuronx_cache.py

```diff
@@ -214,16 +267,16 @@ class ModelCacheEntry:
 
     """
 
-    def __init__(self, model_id: str, config: PretrainedConfig):
+    def __init__(self, model_id: str, config: Union[PretrainedConfig, Dict]):
         self.model_id = model_id
         # Remove keys set to default values
-        self.config = config.to_diff_dict()
+        self.config = config.to_diff_dict() if isinstance(config, PretrainedConfig) else config
```

Comment:
Since `self.config` is changed afterwards by popping some elements, I would recommend working on a copy of it.
```suggestion
        self.config = config.to_diff_dict() if isinstance(config, PretrainedConfig) else dict(config)
```